src/Nfts/ImageGallery/ImageGallery.tsx

```diff
@@ -58,81 +88,159 @@ interface ModeratedImageProps {
+    </>
+  )
+}
+
+function RequiresConsentNFT({uri, text}: {text: string; uri: string}) {
   return (
-    <SkeletonPlaceholder enabled={true}>
-      <View style={styles.imageContainer}>
-        <View style={styles.image} />
-        <SkeletonPlaceholder.Item style={styles.textTop} marginTop={8} marginBottom={13}>
-          <Text style={styles.textTop}>Loading...</Text>
-        </SkeletonPlaceholder.Item>
+    <View>
+      <View style={styles.imageWrapper}>
+        <Image source={{uri}} style={[styles.image, {width: imageSize, height: imageSize}]} blurRadius={20} />
+        <View style={styles.eyeWrapper}>
+          <Icon.EyeOff size={20} color="#FFFFFF" />
+        </View>
       </View>
-    </SkeletonPlaceholder>
+      <Spacer height={IMAGE_PADDING} />
+      <Text style={[styles.textTop, {width: imageSize}]}>{text}</Text>
+      <Spacer height={TEXT_PADDING} />
+    </View>
+  )
+}
+
+function ApprovedNFT({uri, text}: {text: string; uri: string}) {
+  return (
+    <>
+      <Image source={{uri}} style={[styles.image, {width: imageSize, height: imageSize}]} />
+      <Spacer height={IMAGE_PADDING} />
+      <Text style={[styles.textTop, {width: imageSize}]}>{text}</Text>
+      <Spacer height={TEXT_PADDING} />
+    </>
+  )
+}
+
+function SkeletonImagePlaceholder() {
+  return (
+    <View
+      style={[
+        styles.imageContainer,
+        {width: imageSize + 10, height: imageSize + IMAGE_PADDING + TEXT_SIZE + TEXT_PADDING},
+      ]}
+    >
+      <SkeletonPlaceholder enabled={true}>
+        <View>
+          <View style={{width: imageSize, height: imageSize, borderRadius: 8}} />
+          <View
+            style={{
+              marginTop: TEXT_PADDING,
+              width: (imageSize * 3) / 4,
+              height: TEXT_SIZE,
+              borderRadius: 8,
+            }}
+          />
+        </View>
+      </SkeletonPlaceholder>
+    </View>
   )
 }
 
 const styles = StyleSheet.create({
-  safeAreaView: {
-    flex: 1,
-    backgroundColor: 'white',
+  imageWrapper: {
+    position: 'relative',
+  },
+  eyeWrapper: {
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    bottom: 0,
+    right: 0,
+    zIndex: 1,
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
   },
   galleryContainer: {
     flex: 1,
-    justifyContent: 'center',
-    alignItems: 'center',
+    justifyContent: 'flex-start',
+    alignItems: 'flex-start',
     flexDirection: 'row',
     flexWrap: 'wrap',
   },
   imageContainer: {
     paddingHorizontal: 5,
   },
   image: {
-    height: 175,
-    width: 175,
+    height: 164,
+    width: 164,
     borderRadius: 8,
   },
   textTop: {
-    height: 16,
-    width: 148,
-    borderRadius: 50, // skeleton styling
+    fontSize: 14,
+    lineHeight: 22,
+  },
+  row: {
+    display: 'flex',
+    alignItems: 'flex-start',
+    justifyContent: 'space-between',
+    flexDirection: 'row',
   },
 })
+
+const IMAGE_PADDING = 8
+const TEXT_PADDING = 13
```

Comment:
prefer even numbers

Reply:
Yeah and 13 is unlucky, I'll change to 14